### PR TITLE
ws_log() reimplementation for newline character appending

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Please review carefully.

### CODE REVIEW

File: src/logger/module.c [hunk 1]
```diff
@@ -84,30 +84,36 @@ ws_log(
     char* fmt,
     ...
 ) {
-    char* __fmt = fmt;
     va_list list;
-    va_start(list, fmt);
+    size_t fmt_offset = 0;
+    size_t buf_len = 0;
+    size_t pref_len = 0;
```
Comment (neithernut) [inline src/logger/module.c:88]: This can and should be moved to line 95 (into the `if`). Or, optionall, remove `fmt_offset`. Having both is unneccessary.

Comment (matthiasbeyer) [inline src/logger/module.c:88]: Nope. It is used in another scope as well. (see line  106)

Comment (neithernut) [inline src/logger/module.c:88]: Yes, but you can use `pred_len` there, too.
In your implementation, _both_ vars have the same value. It doesn't make sense to have both of them.

**Comment by neithernut** [comment]
ACK.

**Pull request merged**